core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/worker/WorkflowWorker.scala

```diff
@@ -108,12 +108,12 @@ class WorkflowWorker(identifier: ActorVirtualIdentity, operator: IOperatorExecut
     outputCount
   }
 
-  def getResultTuples(): mutable.MutableList[ITuple] = {
+  def getResultTuples(): Option[List[ITuple]] = {
```

Comment:
for containers, is it better to use an empty container indicating no results? just a thought. if use `Option` then the caller would assume `Some(list)` to have some elements.

Reply:
use option is better: this is intentional to distinguish a sink operator vs normal operator. A normal operator doesn't have any result tuples. A sink operator has result tuples, which could be an empty list at a momement.